### DISCUSSION TITLE
Rebase a branch onto another branch from a specific commit

### DISCUSSION THREAD

**Discussion by ekans** [jesseduffield/lazygit#2245] [Q&A] [answered]
When I work on 2 branches (b1, b2) and b2 is based on b1, I use to rebase b2 on b1 when b1 is updated.
In lazygit it's possible to rebase b2 on b1 but if a b1's commit has been modified I like to rebase b2 on b1 like this:
(b2 adds 2 new commits after b1's commits)
```
$ git checkout b2
$ git rebase --onto b1 HEAD~2
```

To be clear, I would like to say: rebase this branch from this commit onto this branch.
Is it doable with lazygit?

**Comment by ekans** [comment, marked as answer]
I answer to myself.
Using `git rebase` with `--update-refs` is the solution I was looking for.

**Reply by ekans** [reply]
(just works in lazygit by enabling it by default)

**Reply by akaihola** [reply]
So what are the actual steps to do this?

If I understand correctly, this is the same as "Rebase subset onto" in Magit.

**Reply by stefanhaller** [reply]
Navigate to the commit below the first one that you want to include in the rebase; hit `shift-B`; select the branch that you want to rebase onto, hit `r`.

**Reply by paulora2405** [reply]
Just to add to this, you can use `git config --global rebase.updateRefs true` to make it the default behavior.
